service/history/execution/context.go

```diff
@@ -1218,6 +1218,13 @@ func (c *contextImpl) updateWorkflowExecutionWithRetry(
 		resp, err = c.shard.UpdateWorkflowExecution(ctx, request)
 		return err
 	}
+	//Preparation for the task Validation.
+	//metricsClient := c.shard.GetMetricsClient()
+	//domainCache := c.shard.GetDomainCache()
+	//executionManager := c.shard.GetExecutionManager()
+	//historymanager := c.shard.GetHistoryManager()
+	//zapLogger, _ := zap.NewProduction()
+	//checker, _ := taskvalidator.NewWfChecker(zapLogger, metricsClient, domainCache, executionManager, historymanager)
```

Comment:
I think we could just guard here using the new flag you created

Reply:
yeah! that's the hope now. I didn't add it because I am not sure if you guys would want to call it here in this flow. currently this seems like a logical place but if you want to go more granular ie. cal the validation only on closed workflows then you might have to move this somewhere else.

We would require to do some extensive testing to finalise the most optimal place to call it.